### PR TITLE
Adding code coverage using jacoco, hoping to add test coverage so tha…

### PR DESCRIPTION
**Pull request by TonyBrobston**
Working on adding a code coverage tool that can fail a build so we can enforce a higher level of coverage. Also a coverage report can be found here after build: target/site/jacoco/index.html

### CODE REVIEW

**Comment by TonyBrobston** [comment]
In an effort to keep coding merging to master, I decreased the minimum coverage required to 0.00 so that we can merge this and I can increase coverage in a future PR. When running `mvn package` with required coverage as 0.01 or greater it currently fails the build.

I'd like to merge this.

I plan to look at adding tests to up the coverage soon.

File: pom.xml [hunk 1]
```diff
@@ -120,22 +125,7 @@
         <dependency>
             <groupId>javax.xml.bind</groupId>
             <artifactId>jaxb-api</artifactId>
-            <version>2.2.11</version>
```
Comment (AndreasChristianson) [inline pom.xml]: does this change go away when https://github.com/dsmHack/2018_team_1_api/pull/39 merges?

File: pom.xml [hunk 2]
```diff
@@ -164,6 +154,54 @@
...
+                                            <minimum>0.00</minimum>
+                                        </limit>
+                                        <limit>
+                                            <counter>BRANCH</counter>
+                                            <value>COVEREDRATIO</value>
+                                            <minimum>0.00</minimum>
+                                        </limit>
+                                    </limits>
+                                    <excludes>
+                                        <exclude>com.xyz.ClassToExclude</exclude>
```
Comment (AndreasChristianson) [inline pom.xml]: looks unneeded

File: pom.xml [hunk 3]
```diff
@@ -164,6 +154,54 @@
...
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <rule>
+                                    <element>CLASS</element>
+                                    <limits>
+                                        <limit>
+                                            <counter>LINE</counter>
+                                            <value>COVEREDRATIO</value>
+                                            <minimum>0.00</minimum>
```
Comment (AndreasChristianson) [inline pom.xml]: Looks like this enforces a min coverage of zero. Should this be higher? Will a higher value prevent ci builds from completing?

Comment (AndreasChristianson) [reply pom.xml]: nm, I see your comment

File: pom.xml (hunk 2 above)
Comment (TonyBrobston) [reply pom.xml]: Ah, good call! I didn't look close enough.

**Pull request merged**